executor/executor.go

```diff
@@ -1251,12 +1251,14 @@ func (e *SelectionExec) Next(ctx context.Context, req *chunk.Chunk) error {
 
 	for {
 		for ; e.inputRow != e.inputIter.End(); e.inputRow = e.inputIter.Next() {
-			if !e.selected[e.inputRow.Idx()] {
-				continue
-			}
 			if req.IsFull() {
```

Comment:
Maybe a more reasonable way is to move the check below `req.AppendRow`?

Reply:
No, logical will be wrong.